experiments/opencog/pattern_matcher_vqa/pattern_matcher_vqa.py

```diff
@@ -345,35 +420,37 @@ def answerOtherQuestion(self, queryInScheme):
                           '%s records, time: %s microseconds',
                           len(resultsData.out), delta.microseconds)
 
-        def extract_predicate(atoms):
-            for atom in atoms:
-                if atom.type == opencog.atomspace.types.InheritanceLink:
-                    if atom.out[1].name != 'BoundingBox':
-                        predicate_name = atom.out[0].name
-                        return predicate_name
+        results = self.sort_results(resultsData, a_extract_predicate=True)
+        if not results:
+            return None, None, None
+        maxResult = results[0]
+        answer = maxResult.get_answer()
+        bb_id = maxResult.get_bounding_box_id()
+        expression = maxResult.get_expression()
+        return answer, bb_id, expression
 
+    def sort_results(self, resultsData, a_extract_predicate=False):
         results = []
         for resultData in resultsData.out:
             out = resultData.out
             if resultData.type == opencog.atomspace.types.AndLink:
                 # resultData is AndLink with random order of conjucts
                 strength = resultData.tv.mean
                 confidence = resultData.tv.confidence
-                predicate_name = extract_predicate(out)
-                if not predicate_name:
-                    continue
-                results.append(OtherDetSubjObjConjunction(strength, confidence, predicate_name))
+                bounding_box_id = extract_bb_id(out)
+                if a_extract_predicate:
+                    predicate_name = extract_predicate(out)
+                    if not predicate_name:
+                        continue
+                else:
+                    predicate_name = None
+                results.append(ConjunctionResult(strength, confidence, predicate_name, bounding_box_id, resultData))
```

Comment:
I would propose keep only resultData in ConjunctionResult and calculate strength, confidence, predicate_name and bounding_box_id from resultData on the fly.
```python
class ConjunctionResult(OtherDetSubjObj):
    def __init__(resultData):
        self.resultData = resultData

    # getter or property
    def strength(self):
        return resultData.tv.mean
```